components/automate-ui/src/app/modules/infra-proxy/create-environment-modal/create-environment-modal.component.html

```diff
@@ -0,0 +1,109 @@
+<chef-modal [visible]="visible" (closeModal)="closeCreateModal()">
+  <div class="text-right">
+    <h2 slot="title">Create Environment</h2>
+  </div>
+  <div class="flex-container">
+    <div class="main pages-body">
+      <app-infra-tab-change class="tabs-row" (tabChange)="tabChange($event)">
+        <app-infra-tab tabTitle="Details" [active]="detailsTab" [disabled]="false">
+          <form [formGroup]="detailsFormGroup">
+            <div class="input-margin">
+              <chef-form-field>
+                <label>
+                  <span class="label">Name <span aria-hidden="true">*</span></span>
+                  <input chefInput formControlName="name" type="text" id="name-input" (keyup)="handleInput($event)" autocomplete="off"/>
+                </label>
+                <chef-error
+                  *ngIf="(detailsFormGroup.get('name').hasError('required') || detailsFormGroup.get('name').hasError('pattern')) && detailsFormGroup.get('name').dirty">
+                  Display Name is required.
+                </chef-error>
+                <chef-error *ngIf="conflictError">
+                  Environment "{{detailsFormGroup.get('name').value}}" already exists.
+                </chef-error>
+              </chef-form-field>
+            </div>
+            <div class="input-margin">
+              <chef-form-field>
+                <label>
+                  <span class="label">Description <span aria-hidden="true">*</span></span>
+                  <textarea chefInput name="description" id="description" formControlName="description" autofocus></textarea>
+                </label>
+                <chef-error
+                  *ngIf="(detailsFormGroup.get('description').hasError('required') || detailsFormGroup.get('description').hasError('pattern')) && detailsFormGroup.get('description').dirty">
+                  Description is required.
+                </chef-error>
+              </chef-form-field>
+            </div>
+            <div class="button-bar-box">
+              <chef-button tertiary (click)="closeCreateModal()">Cancel</chef-button>
+              <chef-button [disabled]="!detailsFormGroup?.valid || conflictError"  primary id="create-button-object-modal" (click)="createEnvironment()">
+                <chef-loading-spinner *ngIf="creating"></chef-loading-spinner>
+                <span *ngIf="!creating">Create</span>
+                <span *ngIf="creating">Creating...</span>
+              </chef-button>
+            </div>
+          </form>
+        </app-infra-tab>
+        <app-infra-tab tabTitle="Constraints" [active]="constraintsTab" [disabled]="!detailsFormGroup?.valid">
+          <div *ngIf="showConstraint">
+            <app-infra-environment-constraint
+              [constraintKeys]="constraintKeys"
+              (constraintValues)="constraintItemsHandler($event)"
+              [dynamicArray]="dynamicArray"
```

Comment:
As I mentioned last time, I would further suggest renaming "dynamicArray" on both sides; that conveys no meaning to the reader. Should convey what is in the array or what it is used for.

Reply:
changed dynamic array to cookbook version name . Updated changes